community/kernel/src/main/java/org/neo4j/kernel/api/cursor/PropertyItem.java

```diff
@@ -43,26 +41,56 @@
     boolean booleanValue();
 
     /**
+     * @return the long value of the current property
+     * @throws IllegalStateException if current property is not a long number
+     */
+    byte byteValue();
+
+    /**
+     * @return the long value of the current property
+     * @throws IllegalStateException if current property is not a long number
+     */
+    short shortValue();
+
+    /**
      * @return the integer value of the current property
-     * @throws IllegalStateException if current property is not an integer number
+     * @throws IllegalStateException if current property is not a integer number
+     */
+    int intValue();
+
+    /**
+     * @return the long value of the current property
+     * @throws IllegalStateException if current property is not a long number
      */
     long longValue();
 
     /**
-     * @return the real value of the current property
-     * @throws IllegalStateException if current property is not a real number
+     * @return the float value of the current property
+     * @throws IllegalStateException if current property is not a float number
+     */
+    float floatValue();
+
+    /**
+     * @return the double value of the current property
+     * @throws IllegalStateException if current property is not a double number
      */
     double doubleValue();
 
     /**
+     * @return the char value of the current property
+     * @throws IllegalStateException if current property is not a char
+     */
+    char charValue();
+
+    /**
      * @return the string value of the current property
+     * @throws IllegalStateException if current property is not a string
      */
     String stringValue();
 
     /**
-     * The byte array data of the current value.
-     *
-     * @param channel to write the data into
+     * @return the array value of the current property
+     * @throws IllegalStateException if current property is not an array
      */
-    void byteArray( WritableByteChannel channel );
+    Object arrayValue();
```

Comment:
No. It probably violates the type system we are moving to so I am gonna make it private.